base_multi_company/models/multi_company_abstract.py

```diff
@@ -0,0 +1,37 @@
+# -*- coding: utf-8 -*-
+# Copyright 2017 LasLabs Inc.
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl.html).
+
+from odoo import api, fields, models
+
+
+class MultiCompanyAbstract(models.AbstractModel):
+
+    _name = 'multi.company.abstract'
+    _description = 'Multi-Company Abstract'
+
+    company_id = fields.Many2one(
+        string='Company',
+        comodel_name='res.company',
+        compute='_compute_company_id',
+        store=True,
```

Comment:
Ok I just tried on Runbot, and it looks like the `name_get` is not a problem. There is a security issue if you click the company though. Take a look at Agrolait on the [v9 Runbot](http://3286807-9-0-7f3092.runbot1.odoo-community.org/) under demo/demo user. 

There is still an inconsistency when searching `company_id` for a company that isn't the primary though. Not sure how often that's actually used though.